src/tests/ftest/util/general_utils.py

```diff
@@ -605,17 +605,25 @@ def stop_processes(hosts, pattern, verbose=True, timeout=60):
     result = {}
     log = getLogger()
     log.info("Killing any processes on %s that match: %s", hosts, pattern)
+
+    if added_filter:
+        ps_cmd = "ps x | grep -E {} | grep -vE {}".format(pattern, added_filter)
+    else:
+        pc_cmd = "pgrep --list-full {}".format(pattern)
+
     if hosts is not None:
         commands = [
             "rc=0",
-            "if pgrep --list-full {}".format(pattern),
+            "set -x",
+            "ps aux",
+            "if " + ps_cmd.format(pattern),
```

Comment:
With the pattern already applied to the `ps_cmd` definition, this could be simplified:
```suggestion
            "if " + ps_cmd,
```

Reply:
1b6386dc60654a50bf68dddacfb23452625090e0